0x06c - iOS tracing section updated (#1834)
OWASP/owasp-mstg: Document/0x06c-Reverse-Engineering-and-Tampering.md

### Tracing

> Tracing involves recording the information about a program's execution. In case of iOS, unlike Android, there are limited options available for tracing various aspects of a program. In this section we will be heavily relying on tools like Frida for performing tracing.

Reviewer: ```suggestion
Tracing involves recording the information about a program's execution. In contrast to Android, there are limited options available for tracing various aspects of an iOS app. In this section we will be heavily relying on tools such as Frida for performing tracing.
```

 21324 ms     | -[NSURLRequest initWithURL:0x106388b00 cachePolicy:0x0 timeoutInterval:0x106388b80
```

#### Native Libraries Tracing

> As discussed earlier in this chapter, iOS applications can also contain native code (C/C++ code) and it can be traced using the `frida-trace` CLI as well. For example, to trace calls for `open` function, following command can be used:

Reviewer: ```suggestion
As discussed earlier in this chapter, iOS applications can also contain native code (C/C++ code) and it can be traced using the `frida-trace` CLI as well. For example, you can trace calls to the `open` function by running the following command:
```

> ```

Reviewer: ```suggestion
```bash
```

$ frida-trace -U -i "open" sg.vp.UnCrackable1
```

> The overall approach and further improvisation for tracing native code using Frida is similar to Android, discussed in "[Tracing](0x05c-Reverse-Engineering-and-Tampering.md#tracing "Tracing")" section.

Reviewer: ```suggestion
The overall approach and further improvisation for tracing native code using Frida is similar to the one discussed in the Android "[Tracing](0x05c-Reverse-Engineering-and-Tampering.md#tracing "Tracing")" section.
```

> On iOS, there are no tools like `strace` or `ftrace` available to trace syscalls or function calls of a program. While `DTrace`, a very powerful and versatile tracing tool, is available for MacOS, but not for iOS.

Reviewer: ```suggestion
Unfortunately, there are no tools such as `strace` or `ftrace` available to trace syscalls or function calls of an iOS app. Only `DTrace` exists, which is a very powerful and versatile tracing tool, but it's only available for MacOS and not for iOS.
```
